lib/auth-middleware/auth_middleware/__init__.py

```diff
@@ -245,31 +284,146 @@ def verify_userauth(id_token, group_name=None, internal_token_only=False):
-            id_info = jwt.decode(id_token, key=pub_key, audience=target_client_ids, verify=True)
-        except jwt.exceptions.PyJWTError as jwte:
-            logger.warning("jwt error on decode. message = %s" % jwte)
+            id_info = jwt.decode(
+                id_token,
+                phone_verify_secret,
+                audience=phone_verify_audience,
+                verify=True
+            )
+        except jwt.DecodeError as de:
+            logger.warning("error on id_token decode. Message = %s" % str(de))
+            raise OAuthProblem('Invalid token')
+        # import pprint; pprint.pprint(id_info)
+    else:
+        issuer = unverified_payload.get('iss')
+        if not issuer:
+            logger.warning("Issuer not found. Aborting.")
             raise OAuthProblem('Invalid token')
-        if not id_info:
-            logger.warning("id_info was not returned from decode")
+        kid = unverified_header.get('kid')
+        if not kid:
+            logger.warning("kid not found. Aborting.")
             raise OAuthProblem('Invalid token')
+        valid_issuer = False
+        keyset = None
+        target_client_ids = None
+
+        SHIB_HOST = os.getenv('SHIBBOLETH_HOST', '')
+
+        if issuer == 'https://' + SHIB_HOST:
+            if internal_token_only:
+                logger.warning("incorrect token type")
+                raise OAuthProblem('Invalid token')
+            valid_issuer = True
+            keyset = get_keyset(
+                'https://' + SHIB_HOST + '/idp/profile/oidc/keyset')
+            target_client_ids = re.split(
+                ',', (os.getenv('SHIBBOLETH_CLIENT_ID', '')).replace(" ", ""))
+
+        # Comment about the next bit. The Py JWT package's support for getting the keys
+        # and verifying against said key is (like the rest of it) undocumented.
+        # These calls may therefore change without warning without notification in future
+        # releases of that library. If this stops working, check the Py JWT libraries first.
+        if not valid_issuer:
+            logger.warning("invalid issuer = %s" % issuer)
+            raise OAuthProblem('Invalid token')
+
+        id_info = decode_id_token(id_token, keyset, target_client_ids, kid)
     # Store ID info for future references in the current request context.
     g.user_token_data = id_info
-
+    g.user_token = id_token
     return id_info
 
+def verify_userauth_coretoken(group_name=None):
+    id_token = get_bearer_token(request)
+    if not id_token:
+        raise OAuthProblem('Missing id token')
+
+    # check which token auth verify to use
+    unverified_header, unverified_payload = get_unverified_header_payload(
+        id_token)
+
+    isAnonymous = unverified_payload.get('anonymous')
+    if isAnonymous == True:
+        return verify_core_token(group_name)
+    return verify_core_userauth(id_token, group_name)
+
+
 
 def use_security_token_auth(func):
     func._use_security_token_auth = True
```

Comment:
This deserves a comment answering why this is always set to True.

Reply:
Looks like a decorator - maybe a DocString mentioning when/where it should be used.